### ISSUE TITLE
Store Data Migration tries to overwrite default category

### ISSUE THREAD

**Issue by DreLincolin** [magento/data-migration-tool#168]
I do understand that this tool is supposed to move data, but with no data checks really?
Since when you install a new Magento 2 store you get autogenerated the Default Store and Store View with store code 0
Is it not logical that migration should not try to override or insert another store with code 0 ?
You cant have 2 default stores with code 0 or in general the table schema states that store_id field is a Primary Key so giving a value to it is ambiguous by itself
Second thing it's values are set to be Unique values
Third it's an AUTOGEN field -> this will trow an error in any case

**Comment by victor-v-rad** [comment]
In some cases the tool takes into account default Magento 2 DB data rather then just to copy it from Magento 1. Please provide exact conditions/steps of the issue if you have one.

**Comment by DreLincolin** [comment]
How does this some cases occur?
As in i see it as an issue that migration tool tries to overwrite the default all ready existing default store. 
Mostly taking into account the database layout with foreign keys and unique fields, indexes as auto increment.
That's a 99.99 % chance that an error will be triggered and migration fail, and if bulk is set to more than 1 will show a totally different error, that will miss lead you to solve the issue.

**Comment by DreLincolin** [comment]
Conditions
Just have a fresh Magento 2 install ( which will have a default store group, a default store, and a default website) that are generated by default in the installation process.

And have a Magento 1.7 store - can be also a fresh install, since it will generate the same default store.

Try and migrate data, on the store data migration step it will throw an error.

simple as that

**Comment by DreLincolin** [comment]
I managed to solve that but with big hastleI had to maualy delete the Default store in the database, with all it's group and website.

Migrate the data.

After had to change the foreign key on update to Cascade ( Restrict by default)
Add the default store back, store group and website - if not this results in the error that website not found.

In some cases, the Cascade on Update wont let you add or change data. So you will have to delete the foreign keys also. And after recreate them.

Hope this is helpful for anyone stuck on this step

**Comment by victor-v-rad** [comment]
Hi @DreLincolin 
I see you have reported many issues recently. Are they related to one DB of Magento 1 you trying to migrate? If so I would be glad to get it and try to migrate to reveal them and fix. You can send the db (remove all the passwords before) through the email (find it in my profile https://github.com/victor-v-rad)

**Comment by DreLincolin** [comment]
Wanted to write in russian language, but i think this wont be a relevant convo then.
yes i tried actually a couple of databases of existing stores. And had to solve each time same problems. Which could have been avoided.

i cannot supply the databases, since i'm not allowed to do so. Since there is a lot of confidential data, like credit card data, email accounts, home address of customers.

I did manage to finish the data migration process on my own, so thank you that is solved. I just thought i report the errors that seemed major to me. And also i'm sure i'm not the only one stuck on this steps. I did actually waste a lot of time by google-ing and there really is slacky info and help. Guess because Magento 2 is still a fresh thing.

I can actually supply a magento store database of a less of importance data that it holds.
